### PR TITLE
feat(terraform) adding trivy:ignore for tf scans to align with trivy

### PR DESCRIPTION
**Pull request by giorod3**
Aligning with trivy to allow inline ignores using trivy:ignore.

### CODE REVIEW

File: pkg/scanners/terraform/scanner_test.go [hunk 1]
```diff
@@ -66,6 +66,24 @@ resource "something" "else" {}
...
+	}
+	results := scanWithOptions(t, `
+//trivy:ignore:altid
+resource "something" "else" {}
+`, options...)
+	require.Len(t, results.GetFailed(), 0)
+	require.Len(t, results.GetIgnored(), 1)
+
+}
+
```
Comment (simar7) [inline pkg/scanners/terraform/scanner_test.go:86]: As most of the other inputs are the same, can we make this into a table driven test instead? So something like:

testCases := []struct{
name string
inputOptions string}{
{ name: "tfsec", inputOptions: "..."}}

then you can iterate over them as inputs:

for _, tc := range testCases {
t.Run(...)
}

File: test/trivy_ignore_test.go [hunk 2]
```diff
@@ -0,0 +1,393 @@
+package test
```
Comment (simar7) [inline test/trivy_ignore_test.go:1]: Likewise with this file as well. `ignore_test.go` could include all the ignores, trivy and tfsec as test inputs for each of the existing tests within `ignore_test.go`.

File: test/trivy_ignore_test.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	"github.com/aquasecurity/defsec/pkg/severity"
+
+	"github.com/aquasecurity/defsec/pkg/scan"
+
+	"github.com/aquasecurity/defsec/internal/rules"
+
+	"github.com/aquasecurity/defsec/pkg/providers"
+
+	"github.com/stretchr/testify/assert"
+)
```
Comment (simar7) [inline test/trivy_ignore_test.go:18]: lint:

```suggestion
	"fmt"
	"testing"

	"github.com/aquasecurity/defsec/internal/rules"
	"github.com/aquasecurity/defsec/pkg/providers"
	"github.com/aquasecurity/defsec/pkg/scan"
	"github.com/aquasecurity/defsec/pkg/severity"
	"github.com/aquasecurity/defsec/pkg/terraform"
	"github.com/stretchr/testify/assert"
)
```

If you're IntelliJ, you can setup `gofmt` to run as a file watcher ever time you make a change.

**Comment by giorod3** [comment]
@simar7 I made the changes we discussed. please review and merge if all looks good

**Review by simar7 — APPROVED** [review]

**Pull request merged**